x/evm/keeper/state_transition.go

```diff
@@ -181,6 +184,16 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 
 	res.Hash = txHash.Hex()
 	logs := k.GetTxLogs(txHash)
+
+	if !res.Failed() {
+		err = k.PostTxProcessing(txHash, logs)
+		if err != nil {
+			k.RevertToSnapshot(revision)
+			res.VmError = "failed to process native logs"
+			res.Ret = []byte(err.Error())
```

Comment:
is this how the error is returned on geth (i.e just the byte representation of the error message)?

Reply:
Only `ErrExecutionReverted` put error reason in the `Ret` field, it's encoded in ABI format, other VM errors don't have error reasons.

Reply:
done